### PR TITLE
Test File for Locker Contract

### PR DESCRIPTION
**Pull request by firllyw**

### CODE REVIEW

File: test/Locker.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const tokenHash = keccak256(['bytes'], [pack(['string'], ['Test Token'])]);
+  beforeEach(async () => {
+    // Deploy Locker Contract on each test iteration
+    locker = await (await ethers.getContractFactory('Locker')).deploy();
+    const setupNftGemGovernorResult = await setupNftGemGovernor();
+    NFTGemMultiToken = setupNftGemGovernorResult.NFTGemMultiToken;
+    const [sender] = await ethers.getSigners();
+    // Mint token to sender to make sure the sender have enough balance
+    await NFTGemMultiToken.mint(sender.address, tokenHash, 100);
+    await NFTGemMultiToken.connect(sender).setApprovalForAll(locker.address, true);
```
Comment (sschepis) [inline test/Locker.test.ts]: what's the connect for? do it like this: await NFTGemMultiToken.setApprovalForAll(locker.address, true);

File: test/Locker.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  })
+  it('Drop Off', async function () {
```
Comment (sschepis) [inline test/Locker.test.ts:25]: Why are you picking an edge case for pick up and drop off? You are using the same token hash for both your unlock token and your award token. While this is in theory possible it's not the primary case you should be testing.  Instead, above in your beforeEach, create 2 different token hashes. They can just be 1 and 2 for ease of the test. 1 is just as good as maxuint256 in the test and easier to read and inderstand in a test case.

File: test/Locker.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { pack, keccak256 } from "@ethersproject/solidity";
+import { ethers } from "hardhat";
+import {
+  setupNftGemGovernor
+} from './fixtures/Governance.fixture';
+
+describe('Locker Test Suite', function () {
+  let locker: Contract;
+  let NFTGemMultiToken: any;
+  const tokenHash = keccak256(['bytes'], [pack(['string'], ['Test Token'])]);
```
Comment (sschepis) [inline test/Locker.test.ts]: No need to make a hash. you can just use '1' , '2' etc for different token types

File: test/Locker.test.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    await locker.dropOff(
+      NFTGemMultiToken.address,
+      tokenHash,
+      NFTGemMultiToken.address,
+      tokenHash,
+      5,
+    )
+    await locker.pickUpTokenWithKey(tokenHash);
+    const lockerContent = await locker.contents(tokenHash);
+    expect(lockerContent.awardQty).to.be.equal(0);
```
Comment (sschepis) [inline test/Locker.test.ts]: You should also test whether the item is in your possession here

File: test/Locker.test.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    await NFTGemMultiToken.connect(sender).setApprovalForAll(locker.address, true);
+  })
+  it('Drop Off', async function () {
+    await locker.dropOff(
+      NFTGemMultiToken.address,
+      tokenHash,
+      NFTGemMultiToken.address,
+      tokenHash,
+      5);
+    let lockerContent = await locker.contents(tokenHash);
```
Comment (sschepis) [inline test/Locker.test.ts]: This test has no comments. There are several logical operations here but zero comments as to what is going on. That's not acceptable. Look at it this way: At some point, future Firrly, or someone else, is going to come reading this file. You have the power right now to make that future person's day a little happier by leaving them a piece of code that takes them minimal energy to easily understamd. Or, you can annoy someone in the future by leaving a mess. There is no saving time by omitting comments. Someone elways has to pay and usually it's you.

**Review by sschepis — CHANGES_REQUESTED** [review]
Hey Firrly, take a look at my comments and also please add additional tests to check that unauthorized people can't withdraw Locker contents.  You have access to several accounts within your text context so it's easy to test. Just have account A deposit something and then have account B attempt to withdraw. You can specify which account it's coming from in your tests by adding a parameter to blockchain function calls: `{ from: 'fromaddress'}`

**Comment by firllyw** [comment]
> Hey Firrly, take a look at my comments and also please add additional tests to check that unauthorized people can't withdraw Locker contents. You have access to several accounts within your text context so it's easy to test. Just have account A deposit something and then have account B attempt to withdraw. You can specify which account it's coming from in your tests by adding a parameter to blockchain function calls: `{ from: 'fromaddress'}`

Thanks sir, working on it...
